packages/kbn-apm-synthtrace/src/scenarios/simple_logs.ts

```diff
@@ -9,67 +9,84 @@ import { LogDocument, log, generateShortId, generateLongId } from '@kbn/apm-synt
 
+  const constructLogsCommonData = () => {
+    const index = Math.floor(Math.random() * 3);
+    const serviceName = getServiceName(index);
+    const logMessage = MESSAGE_LOG_LEVELS[index];
+    const cluster = getCluster(index);
+    const cloudRegion = getCloudRegion(index);
+
+    return {
+      index,
+      serviceName,
+      logMessage,
+      cluster,
+      cloudRegion,
+    };
+  };
+
   return {
     bootstrap: async ({ logsEsClient }) => {
       if (isLogsDb) await logsEsClient.createIndexTemplate(IndexTemplateName.LogsDb);
     },
     generate: ({ range, clients: { logsEsClient } }) => {
       const { logger } = runOptions;
 
-      // Logs Data logic
-      const MESSAGE_LOG_LEVELS = [
-        { message: 'A simple log with something random <random> in the middle', level: 'info' },
-        { message: 'Yet another debug log', level: 'debug' },
-        { message: 'Error with certificate: "ca_trusted_fingerprint"', level: 'error' },
-      ];
-      const CLOUD_PROVIDERS = ['gcp', 'aws', 'azure'];
-      const CLOUD_REGION = ['eu-central-1', 'us-east-1', 'area-51'];
-
-      const CLUSTER = [
-        { clusterId: generateShortId(), clusterName: 'synth-cluster-1', namespace: 'default' },
-        { clusterId: generateShortId(), clusterName: 'synth-cluster-2', namespace: 'production' },
-        { clusterId: generateShortId(), clusterName: 'synth-cluster-3', namespace: 'kube' },
-      ];
-
-      const SERVICE_NAMES = Array(3)
-        .fill(null)
-        .map((_, idx) => `synth-service-${idx}`);
-
       const logs = range
         .interval('1m')
         .rate(1)
         .generator((timestamp) => {
           return Array(3)
             .fill(0)
             .map(() => {
-              const index = Math.floor(Math.random() * 3);
-              const { message, level } = MESSAGE_LOG_LEVELS[index];
-              const serviceName = SERVICE_NAMES[index];
+              const {
+                serviceName,
+                logMessage: { level, message },
+                cluster: { clusterId, clusterName, namespace },
+                cloudRegion,
+              } = constructLogsCommonData();
 
               return log
                 .create({ isLogsDb })
                 .message(message.replace('<random>', generateShortId()))
                 .logLevel(level)
                 .service(serviceName)
+                .setGeoLocation(getGeoCoordinate())
+                .setHostIp(getIpAddress())
                 .defaults({
                   'trace.id': generateShortId(),
                   'agent.name': 'nodejs',
-                  'orchestrator.cluster.name': CLUSTER[index].clusterName,
-                  'orchestrator.cluster.id': CLUSTER[index].clusterId,
-                  'orchestrator.namespace': CLUSTER[index].namespace,
-                  'container.name': `${SERVICE_NAMES[index]}-${generateShortId()}`,
+                  'orchestrator.cluster.name': clusterName,
```

Comment:
There's a log of repetition of lines like these. What if `constructLogsCommonData()` returned an object with the proper keys so all these could be replaced with something `...commonLogEntryFields`?